### DISCUSSION TITLE
setting up a franka emika robot

### DISCUSSION THREAD

**Discussion by zvonkovugreshek** [visose/Robots#198] [Q&A] [answered]
Hello,

I have a franka robot and i have suddenly seen that it is available to be controlled by the robots plugin. However I would like to request your help on how to set up the xml/rhino file since i dont see anyone in the libraries to be using it. Any help would be much appreciated.

Best
Z

**Comment by visose** [comment]
Hi @zvonkovugreshek, support for Franka Emika is still pending to be tested, this will happen soon.
Once things are working, the Bartlett robot library will be updated with a Franka Emika Panda, and I will write down in the wiki a bit of info on how to use.
But in summary, the "Save Program" component will output a python script that uses this library: https://github.com/pantor/frankx
This script should be run in a Linux machine (with realtime kernel), conected to the robot using the FCI interface. 
The Remote component is updated so you can SSH and FTP into the linux machine to transfer, run and monitor the program.
You can force the redudant axis to a position by adding an External input to a Create Target component and feeding a single value per target. Some features are not supported.

**Reply by zvonkovugreshek** [reply]
Thanks for the swift reply @visose. I might also help you with the debugging/testing :) I was thinking on fetching some xml's that i have seen from moveit or gazebo configurations and actually wanted to try out the frankx driver but havent had the time to. You should also give a try with Armer if things are not working as expected. I have a rt linux machine with ros and all the components necessary so for me only the rhino link is missing. Perhaps we could collaborate on this one :)

**Reply by visose** [reply]
Great, you have all the necessary hardware. As tempting as it is to offload the testing to you, I'm sure there will be many issues that must be fixed before anyone else can have a go at testing it.

The Bartlett bought first gen Pandas and the controller has since been deprecated and would not support FCI. A replacement has already been shipped and just waiting for it to arrive to start testing.

**Reply by zvonkovugreshek** [reply]
Yeah i totally understand you as a developer. But how would you suggest to start implementing it on my own, i.e using an existing template for the franka or which xml are you using to generate the robot? Mine has the fci enabled and on the linux side it works good. I'm just curious how could i implement and test the workflow in general.

**Comment by visose** [comment, marked as answer]
Closing this as superseded by current Franka support. Franka Emika support now exists in Robots, with a Panda in the Bartlett library, and generated programs use the Frankx/FCI workflow rather than only the Desk interface. See discussion #253 for the setup summary. Please open a fresh discussion with current version details if a specific Franka library or execution step still fails.
